File: ReceiptPrinter/DTOs/FoodItemDto.py
from pydantic import BaseModel
from typing import Optional, List
from .FoodItemOptionDto import FoodItemOptionDto


class FoodItemDto(BaseModel):
    id: int
    name: str = ""
    orderId: Optional[int] = None
    stationId: Optional[int] = None
    saleCardId: Optional[int] = None
    saleSwipeId: Optional[int] = None
    swipeCost: Optional[int] = None
    cardCost: Optional[float] = None
    special: Optional[bool] = False
    options: List[FoodItemOptionDto] = []

File: Cafeteria.Shared/DTOs/Order/FoodItemDto.cs
namespace Cafeteria.Shared.DTOs.Order;

public class FoodItemDto
{
    public int Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public int OrderId { get; set; }
    public int StationId { get; set; }
    public int? SaleCardId { get; set; }
    public int? SaleSwipeId { get; set; }
    public int? SwipeCost { get; set; }
    public decimal? CardCost { get; set; }
    public bool Special { get; set; }
    public List<FoodItemOptionDto> Options { get; set; } = new();
}
